src/js/protocols/stm32usbdfu.js

```diff
@@ -131,14 +131,13 @@ STM32DFU_protocol.prototype.closeDevice = function () {
 
     chrome.usb.closeDevice(this.handle, function closed() {
         if (checkChromeRuntimeError()) {
-            console.log('Failed to close USB device!');
+            console.log(`Failed to close USB device with Handle ID: ${self.handle.handle}`);
             GUI.log(i18n.getMessage('usbDeviceCloseFail'));
+            self.handle = null;
+        } else {
+            GUI.log(i18n.getMessage('usbDeviceClosed'));
+            console.log(`Device closed with Handle ID: ${self.handle.handle}`);
         }
-
-        GUI.log(i18n.getMessage('usbDeviceClosed'));
-        console.log(`Device closed with Handle ID: ${self.handle.handle}`);
-
```

Comment:
It makes no logical sense to say that the USB device was closed, which implies successfully closed, if in fact it failed to close.  So I put these lines after an else, so we get a message saying tha closing failed if it failed, and was successful if it was successful.